gnome-image-installer/pages/finished/gis-finished-page.c

```diff
@@ -134,42 +134,54 @@ toggle_leds (GisPage *page)
 #define EOS_IMAGE_VERSION_ALT_PATH "/"
 
 static gchar *
-get_image_version (const gchar *path)
+gis_page_util_get_image_version (const gchar *path,
+                                 GError     **error)
 {
   ssize_t attrsize;
-  gchar *value;
+  g_autofree gchar *value = NULL;
+  int errsv;
 
   g_return_val_if_fail (path != NULL, NULL);
 
   attrsize = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
-  if (attrsize < 0) {
-    g_message ("Error examining " EOS_IMAGE_VERSION_XATTR " on %s: %s",
-               path, g_strerror (errno));
-    return NULL;
-  }
+  if (attrsize < 0)
+    goto error;
 
   value = g_malloc (attrsize + 1);
   value[attrsize] = 0;
 
   attrsize = getxattr (path, EOS_IMAGE_VERSION_XATTR, value,
                        attrsize);
-  if (attrsize < 0) {
-    g_warning ("Error reading " EOS_IMAGE_VERSION_XATTR " on %s: %s",
-               path, g_strerror (errno));
-    g_free (value);
-    return NULL;
-  }
-
-  return value;
+  if (attrsize < 0)
+    goto error;
+
+  return g_steal_pointer (&value);
+
+error:
+  errsv = errno;
+  g_set_error (error, G_IO_ERROR, g_io_error_from_errno (errsv),
+               "Error examining " EOS_IMAGE_VERSION_XATTR " on %s: %s",
+               path, g_strerror (errsv));
+  return NULL;
```

Comment:
Since we're using `g_autoptr` above, it'd be nice to avoid the goto error stuff here - to avoid the `g_set_error` duplication you can wrap it in a function like `error_about_invalid_xattr (GError **);`, which just sets the error and returns.

Reply:
I don't think adding an extra function improves clarity, but I have eliminated the `goto` in a fixup commit. See what you think. The same code and fix is [present in g-i-s](https://github.com/endlessm/gnome-initial-setup/pull/227/commits/4a9a44d1697109ebde3a08527b2db30ae6832114) so I'll apply the same change there if this is preferable.

Reply:
Fair enough. Its more of a personal preference and not important.

Reply:
I squashed the fixup commit out of habit, sorry :(